src/lib.rs

```diff
@@ -507,6 +508,11 @@ impl Build {
             .debug(false)
             .cpp(self.cpp)
             .cuda(self.cuda);
+
+        if !cfg.has_flags() {
+            cfg.warnings_into_errors(true);
+        }
```

Comment:
I think this is too generous, flags is most likely going to be something like `-O2` than `-Wno-error`. We think we should always apply `-Werror` but just make sure it comes before `CFLAGS`/`CXXFLAGS`.

Reply:
> I think we should always apply -Werror but just make sure it comes before CFLAGS/CXXFLAGS.

I've enabled `-Werror` in any case but I can't put it before `CFLAGS`/`CXXFLAGS` unless I change `Build::try_get_compiler` or do sth. similar to #784